modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -719,21 +719,19 @@ def validate_title_ten_activiation_values!(service_information)
 
         return if title_ten_activation.blank?
 
-        form_obj_desc = 'title 10 activation'
+        form_obj_desc = 'federal activation'
 
         if title_ten_activation_date.blank?
-          raise_exception_if_value_not_present('title 10 activation date',
+          raise_exception_if_value_not_present('federal activation date',
                                                form_obj_desc)
         end
 
-        if anticipated_seperation_date.blank?
-          raise_exception_if_value_not_present('anticipated seperation date',
-                                               form_obj_desc)
-        end
+        return if anticipated_seperation_date.blank?
+
```

Comment:
@FonzMP This is where we were returning an error for a nullable attribute being blank.

Reply:
there's a lot going on with this.... and I'm trying to make sense of it...

first, the method here has a typo. probably should fix that. 
`validate_title_ten_activiation_values`

also, this runs twice, 
https://github.com/department-of-veterans-affairs/vets-api/blob/f5b7fae89cef51063b9256fb2d834b992cd62d68/modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb#L685
and one conditionally required per the comment
https://github.com/department-of-veterans-affairs/vets-api/blob/f5b7fae89cef51063b9256fb2d834b992cd62d68/modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb#L692
`# if reserves is not empty the we require tos dates`


my biggest concern is that inside of reserves, we don't even have a `federalActivation` so we'll never get past the [return](https://github.com/department-of-veterans-affairs/vets-api/blob/f5b7fae89cef51063b9256fb2d834b992cd62d68/modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb#L720C11-L720C11). so that means it's not requiring the ToS if the comment remains true. 

https://github.com/department-of-veterans-affairs/vets-api/blob/f5b7fae89cef51063b9256fb2d834b992cd62d68/modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb#L716



In respect to your concern, this is enforcing a conditionally required check that's either not documented, or should not be enforced. it looks like it's confirming an activationDate if we receive anything in federalActivation besides nil or a blank object. I think that's expected? but we need to find something to confirm that and we can either update the documentation or skip the check.

The latter check is just _if_ anticipatedSeparationDate exists, it's after one of the service dates. this is past the error you were mentioning though.